pinot-core/src/main/java/org/apache/pinot/core/operator/filter/AndFilterOperator.java

```diff
@@ -45,6 +47,28 @@ protected FilterBlock getNextBlock() {
     return new FilterBlock(new AndDocIdSet(filterBlockDocIdSets));
   }
 
+  @Override
+  public boolean canOptimizeCount() {
+    boolean allChildrenCanProduceBitmaps = true;
+    for (BaseFilterOperator child : _filterOperators) {
+      allChildrenCanProduceBitmaps &= child.canProduceBitmaps();
+    }
+    return allChildrenCanProduceBitmaps;
+  }
+
+  @Override
+  public int getNumMatchingDocs() {
```

Comment:
It will just compute zero if that happens, so this is not required for safety, the real question would be why we have an `AndFilterOperator` with zero children. Also, recall that I had handled this case explicitly and you asked me to remove it earlier.

Reply:
My point is that empty/single child should not get into this operator, and we should guard that (constructor should be a better place for this check though)

Reply:
I agree if this class checks this, it should be done in the constructor. Let's fix that another time.